### ISSUE TITLE
Formatting of byte array as XML does not work

### ISSUE THREAD

**Issue by fml2** [squirrel-sql-client/squirrel-sql-code#47]
Hello,

I don't know whether this is an error in the program or whether it works as it should (then this issue should be considered as a feature request).

I have a table with a column that holds XML data. The column type is "BLOB". When I double click on the column I see this (the "Show ASCII as chars" is not checked):

![grafik](https://github.com/user-attachments/assets/9d7bdc65-9848-4ce9-8c0a-53f804539175)

If I activate the "Show ASCII as chars" option, I see this:

![grafik](https://github.com/user-attachments/assets/d65bbb93-73ae-487d-89ba-a11849f6fb55)

One can clearly see that the characters are an XML but they are separated with three spaces.

Clicking on "Reformat Xml/Json" does not do anything, i.e. the data is not displayed as a formatted XML -- this is the problem I'm talking about in this issue.

How is it possible to display the XML contents as formatted XML in this case?

**Comment by t603** [comment]
Hello, fml2, 

when I am facing sometimes such issue, I will make "cast myCol as clob" workaround: 

Regards, Stepan 

![SQuirreLSQL_screenshot_2024-11-04_16-49a](https://github.com/user-attachments/assets/d99cda9c-191b-4242-bc9f-4dd9417795c8)

**Comment by fml2** [comment]
Thank you for the hint! My goal was to be able to see the formatted data without having to execute SQLs, i.e. by directly browsing the table contents.

**Comment by fml2** [comment]
@t603: BTW: Using "cast(... as clob)" does not work for me. I'm on Postgres, if that matters. I used "cast(.... as text)" and it just displayed the hex chars, without spaces in between.

**Comment by gerdwagner** [comment]
Committed to our GIT repositories. Will be available in future snapshots and versions.

Excerpt from change log:

> https://github.com/squirrel-sql-client/squirrel-sql-code/issues/47
>   Table cell data popup dialog: XML/Json Reformatting now works for binary/BLOB data.

**Comment by fml2** [comment]
Thank you @gerdwagner! I will check this as soon as I can and report back.

**Comment by fml2** [comment]
Hello @gerdwagner,

I’ve tried the new version. It did not work for me. The blob was still displayed as hex digits. My database is a Postgres one and the column data type is `bytea`. The column has an XML contens as data.

I’ve also tried the new Excel export feature and found a couple of glitches.
1. Selecting the “First row bold” has no effect (the first row in the created Excel file is not displayed bold)
2. The i-icons for “First row centered” and “First row bold” both display the hint “First row bold is available …”.

Thank you for working on the program!

**Comment by fml2** [comment]
BTW: What does work for me (Postgres) is `select convert_from ( bytes_, 'UTF-8' ) ...`

**Comment by gerdwagner** [comment]
> I’ve tried the new version. It did not work for me. The blob was still displayed as hex digits. My database is a Postgres one and the column data type is bytea. The column has an XML contents as data.

Sorry, I can't reproduce your problem. I tested on Postgres  Version=14.4 with Driver-Version=42.7.3 and column type bytea. Do you have any more details?


> 1. Selecting the “First row bold” has no effect (the first row in the created Excel file is not displayed bold)
> 2. The i-icons for “First row centered” and “First row bold” both display the hint “First row bold is available …”.

Please see the following new comment in https://github.com/squirrel-sql-client/squirrel-sql-code/issues/46 :
https://github.com/squirrel-sql-client/squirrel-sql-code/issues/46#issuecomment-2495573903

**Comment by fml2** [comment]
> Sorry, I can't reproduce your problem

I found out the reason why the contents was not displayed correctly. In my settings, I had the property "Max characters to display in value popup" set to 10000. The contents has the length of about 40000 bytes. Hence not the whole XML was read and could not be formatted properly. I've increased the value and now the XML is formatted as it should.

It there were a sign that the displayed value was truncated it would be cool. But on the other hand, the dialog should not be cluttered.

I will close the issue now. Thank you for the great program!
